torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -509,6 +511,31 @@ std::shared_ptr<SugaredValue> NamedTupleConstructor::call(
   return std::make_shared<SimpleValue>(self);
 }
 
+std::shared_ptr<BuiltinFunction> BuiltinFunction::tryCreate(
+    Symbol symbol,
+    c10::optional<NamedValue> self) {
+  for (const std::shared_ptr<Operator>& op : getAllOperatorsFor(symbol)) {
+    if (!self) {
+      return std::make_shared<BuiltinFunction>(symbol, nullptr);
+    }
+    if (auto index = op->schema().argumentIndexWithName("self")) {
+      std::unordered_map<std::string, TypePtr> type_env;
```

Comment:
Could we just reuse `tryMatchArgument` here? That way this would also work with conversions.

Reply:
Since this is for method invocation where the method should be determined by the self argument, we do not want to apply conversions to self. Methods should only exist if the type matches without conversion. Our previous implementation was too accepting and open to weird bugs here. Furthermore, calling tryMatchArgument modifies the graph, which we do not want to do here because the call has not been made yet. It is possible that the call ends up being invoked multiple times, or not at all.

Reply:
Not applying conversions makes since here. I don't see why we can't use `tryMatchArgument` with no conversions.

It's already the case that we modify the graph for schemas we don't match to; any extra nodes get DCE'd.

Reply:
tryMatchArgument is private, takes a lot of arguments, and would be awkward to expose. It is not something that should be called outside of schema matching. Even just accessing `self` as a Value in this context requires potentially modifying a graph because self is NamedValue (potentially needs to be inserted as a constant). Refactoring all of tryMatchArgument so that it doesn't modify the graph would be difficult. I don't like the fact that we rely on DCE here and I don't want to propagate it to more places. It makes debugging harder.

Reply:
I think it's a little weird to fix inserting unmatched nodes only for this specific case. i don't know how invasive the more general fix will be though.